distribution/lib/Standard/Visualization/0.0.0-dev/src/Scatter_Plot.enso

```diff
@@ -133,23 +134,78 @@ Table.Table.axes self =
 
 ## PRIVATE
 Vector.Vector.point_data : Vector -> Object
-Vector.Vector.point_data self =
-        self.map_with_index <| i-> elem->
-            Json.from_pairs [[X.name,i],[Y.name,elem]]
+Vector.Vector.point_data self  =
+    self.map_with_index <| i-> elem->
+        Json.from_pairs [[X.name,i],[Y.name,elem]]
 
 ## PRIVATE
-json_from_table : Table -> Json
-json_from_table table =
-    data = table.point_data
+bound_data bounds data = case bounds of
+    Nothing -> data
+    _ ->
+        min_x = bounds.at 0
+        min_y = bounds.at 1
+        max_x = bounds.at 2
+        max_y = bounds.at 3
+
+        data.filter <| datum->
+            x = datum.get "x" . unwrap
+            y = datum.get "y" . unwrap
+
+            min_x<=x && x<=max_x && min_y<=y && y<=max_y
+
+type Extreme min_x max_x min_y max_y
+
+## PRIVATE
+limit_data limit data = case limit of
+    Nothing -> data
+    _ -> if (data.length <= limit) then data else
+        b = Vector.Builder.new
+
+        x datum = (datum.get "x").unwrap
+        y datum = (datum.get "y").unwrap
+
+        check_max_y min now = case min of
+            Extreme min_x max_x min_y Nothing -> Extreme min_x max_x min_y now
```

Comment:
Do we have _guards_ for `case_of` statements? If so, I could merge this four functions into a single one. However I haven't found a way to do so right now, so I had to split that into for functions...

Reply:
Like to have a think on this one I see what it is doing but it feels like we should be able to do it in a simpler way.

Reply:
The request behavior (extremes are present) is covered by a test. Maybe we could integrate and refactor later? 

FYI: My integration deadline was yesterday...